local-env-setup.sh

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+darwin_install_pip3_packages() {
+    echo "Installing setuptools grpcio-tools virtualenv"
+    pip3 install setuptools grpcio-tools virtualenv
+    echo "Installing mypy-protobuf"
+    pip3 install --user mypy-protobuf
+}
+
+install_go_packages(){
+        echo "Installing goavro"
+        go get github.com/linkedin/goavro
+        # As we are using bash, we are assuming .bashrc exists. 
+        grep -qxF "export GOPATH='${pwd}/sdks/go/examples/.gogradle/project_gopath'" ~/.bashrc || echo "export GOPATH='${pwd}/sdks/go/examples/.gogradle/project_gopath'" >> ~/.bashrc
+        # The following source command will not work if executed in a subshell like $ ./this_script.sh. Instead, we need to run it in the current shell as $ . ./this_script.sh.
+        # However, this is just to load the GOPATH env variable from the previosu command into the current shell. If that's not required, then we can suggest a terminal reload after the execution?
+        #source ~/.bashrc
+}
+
+# Running on Linux
+if [ "$(uname -s)" = "Linux" ]; then
+    # Assuming Debian based Linux and the prerequisites in https://beam.apache.org/contribute/ are met:
+    apt-get update
+
+    echo "Installing openjdk-8-jdk, python-setuptools, python3-pip, tox, docker.io"
+    apt-get install -y openjdk-8-jdk python-setuptools python3-pip tox docker.io # using python3-pip as Python3 is required.
+
+    type -P python3 > /dev/null 2>&1
+    python3Exists=$?
+    type -P pip3 > /dev/null 2>&1
+    pip3Exists=$?
+    if [ $python3Exists -eq 0  -a $pip3Exists -eq 0 ]; then
+        echo "Installing grpcio-tools and mypy-protobuf"
+        pip3 install grpcio-tools mypy-protobuf virtualenv
+    else
+        echo "Python3 and pip3 are required. Installation Failed."
```

Comment:
These were installed earlier on line 41 right? That means the installation failed for some reason. Maybe a bit more info in the message would be helpful:

```suggestion
        echo "Python3 and pip3 are required but failed to install. Install them manually and rerun the script."
```

Reply:
Line 41 installed pip3 and other tools, then we use pip3 to install additional libraries with it.